Add rendering module

PepperCode1 opened the pull request:
Highlights:
- Model submodule:
  - Injected interface into `BakedModelManager` for getting `BakedModel`s from `Identifier`s
  - Transitive access wideners for registering item model predicates
- Rendering registration submodule:
  - Dynamic item rendering
  - Custom armor rendering
  - Allow mapping custom `EntityModelLayer`s to `TexturedModelData`s
  - Transitive access wideners for registering block entity and entity renderer factories
  - Move block render layer map from block extensions submodule to rendering registration submodule
  - Allow `ArmorMaterial#getName` to return a namespaced texture location

To do:
- [ ] Add block and item color provider registration
- [ ] Allow registering entity feature renderers
- [ ] Improve javadoc
- [ ] Improve testmods

PepperCode1 committed:
Start work on rendering module

OroArmor reviewed (commented):
looks good so far

LambdAurora commented on library/rendering/registration/src/main/java/org/quiltmc/qsl/rendering/registration/impl/client/DynamicItemRendererRegistryImpl.java:
```diff
@@ -0,0 +1,56 @@
+
+	public static void register(ItemConvertible item, DynamicItemRenderer renderer) {
+		Objects.requireNonNull(item, "item is null");
+		Objects.requireNonNull(item.asItem(), "item is null");
```
Shouldn't ItemStack already disallows this?

PepperCode1 replied:
`ItemStack`? `ItemConvertable` is an interface and could return null.

LambdAurora commented on library/rendering/registration/src/main/java/org/quiltmc/qsl/rendering/registration/impl/client/BlockRenderLayerMapImpl.java:
```diff
@@ -14,7 +14,7 @@
  */
 
-package org.quiltmc.qsl.block.extensions.impl.client;
+package org.quiltmc.qsl.rendering.registration.impl.client;
```
idk how to feel about this

but if it's a move the copyright year still shouldn't be changed

PepperCode1 replied:
The license changes have been reverted.

PepperCode1 committed:
Add EntityModelLayerRegistry and ArmorRenderer

- Remove DynamicItemRendererRegistry
- Remove QuiltArmorMaterial in favor of getName returning a namespaced string
- Revert license changes to block render layer map classes
